src/ptf/mask.py

```diff
@@ -38,7 +44,12 @@ def set_do_not_care_packet(self, hdr_type, field_name):
             ]  # build & parse packet to be sure all fields are correctly filled
         except Exception:  # noqa
             self.valid = False
-            return
+            raise MaskException("Can not build or decode Packet")
+
+        if field_name not in [x.name for x in fields_desc]:
+            self.valid = False
+            raise MaskException("Field %s does not exists in frame" % field_name)
```

Comment:
typo: `Field %s does not exist in frame`

Reply:
Thx 👍